webapp/channels/src/components/channel_layout/channel_controller.tsx

```diff
@@ -40,14 +45,18 @@ export default function ChannelController(props: Props) {
     }, []);
 
     useEffect(() => {
-        const loadStatusesIntervalId = setInterval(() => {
-            dispatch(loadStatusesForChannelAndSidebar());
-        }, Constants.STATUS_INTERVAL);
-
-        return () => {
-            clearInterval(loadStatusesIntervalId);
-        };
-    }, []);
+        if (enableUserStatuses) {
+            const loadStatusesIntervalId = setInterval(() => {
+                dispatch(loadStatusesForChannelAndSidebar());
+            }, Constants.STATUS_INTERVAL);
+
+            return () => {
+                clearInterval(loadStatusesIntervalId);
+            };
+        }
+        return () => {}; // Return a no-op function when enableUserStatuses is false
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [enableUserStatuses]);
```

Comment:
If this is false you should cancel any ongoing requests

Reply:
ok, got it. Thanks